app/src/main/java/org/oppia/app/player/audio/AudioFragmentPresenter.kt

```diff
@@ -99,44 +97,36 @@ class AudioFragmentPresenter @Inject constructor(
       it.audioFragment = fragment as AudioFragment
       it.lifecycleOwner = fragment
     }
-    subscribeToProfileLiveData()
+    subscribeToAudioLanguageLiveData()
     return binding.root
   }
 
-  private val profileLiveData: LiveData<Profile> by lazy {
-    getProfileData()
-  }
-
-  private fun getProfileData(): LiveData<Profile> {
+  private fun getProfileData(): LiveData<String> {
     return Transformations.map(
       profileManagementController.getProfile(profileId),
       ::processGetProfileResult
     )
   }
 
-  private fun subscribeToProfileLiveData() {
-    profileLiveData.observe(activity, Observer<Profile> { result ->
-      audioLanguage = result.audioLanguage
-      languageSelected(getAudioLanguage(audioLanguage))
+  private fun subscribeToAudioLanguageLiveData() {
+    getProfileData().observe(activity, Observer<String> { result ->
+      languageSelected(result)
     })
   }
 
   /** Gets language code by [AudioLanguage]. */
-  fun getAudioLanguage(audioLanguage: AudioLanguage): String {
+  private fun getAudioLanguage(audioLanguage: AudioLanguage): String {
     return when (audioLanguage) {
-      AudioLanguage.NO_AUDIO -> "No Audio"
       AudioLanguage.ENGLISH_AUDIO_LANGUAGE -> "en"
       AudioLanguage.HINDI_AUDIO_LANGUAGE -> "hi"
       AudioLanguage.FRENCH_AUDIO_LANGUAGE -> "fr"
       AudioLanguage.CHINESE_AUDIO_LANGUAGE -> "zh"
-      else -> "No Audio"
+      else -> "en"
     }
   }
 
-  private fun processGetProfileResult(profileResult: AsyncResult<Profile>): Profile {
-    if (profileResult.isFailure()) {
-    }
-    return profileResult.getOrDefault(Profile.getDefaultInstance())
+  private fun processGetProfileResult(profileResult: AsyncResult<Profile>): String {
+    return getAudioLanguage(profileResult.getOrDefault(Profile.getDefaultInstance()).audioLanguage)
```

Comment:
It might still be worth logging a statement in the error case since we don't expect an error to occur here.

Reply:
Done.